pkgs/tools/misc/dgoss/default.nix

```diff
@@ -0,0 +1,57 @@
+{
+  # packaging dependencies
+  lib,
+  fetchFromGitHub,
+  resholvePackage,
+  substituteAll,
+  # runtime dependencies
+  bash,
+  coreutils,
+  goss,
+  which,
+}:
+
+resholvePackage rec {
+  pname = "dgoss";
+  version = "0.3.16";
+
+  src = fetchFromGitHub {
+      owner = "aelsabbahy";
+      repo = "goss";
+      rev = "v${version}";
+      sha256 = "1m5w5vwmc9knvaihk61848rlq7qgdyylzpcwi64z84rkw8qdnj6p";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p "$out"/bin
+    install extras/dgoss/dgoss "$out"/bin/dgoss
+  '';
+
+  solutions = {
+    default = {
+      scripts = [ "bin/dgoss" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = [ coreutils which ];
+      prologue = [
+        (substituteAll {
+          src = ./set-goss-path.sh;
+          inherit goss;
+        })
+      ];
+      fake = {
+        external = [ "docker" ];
+      };
+    };
+  };
```

Comment:
What is this doing? Why are we not just using echo to write that one file?

Reply:
> What is this doing?

I'm not sure I understand the scope of "this" in your question. Do you mean the `set-goss-path.sh` file ? It just sets an environment variable to the path of `goss` binary.

> Why are we not just using echo to write that one file?

I'm not sure I understand your question. Do you mean the `set-goss-path.sh` file ? I had not thought about using echo to write it, I'll try

Reply:
I replaced the `prologue` attribute with running `sed` command to add the environment variable. Is that better ?
I don't know if it was the right thing to do but I added it in the install phase. Please let me know if that was a mistake.